File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/eventbus/MessageProducer.php
<?php 
namespace io\vertx\php\core\eventbus;
class MessageProducer
{
    
    private function __construct()
    {

    }
    /**
     * ()
     * @return string
     * methodSize : 1
     */
    public function address(/*0*/)
    {

    }
    /**
     * ()
     * @return Void
     * methodSize : 1
     */
    public function close(/*0*/)
    {

    }
    /**
     * (DeliveryOptions options)
     * @return MessageProducer
     * methodSize : 1
     */
    public function deliveryOptions(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return MessageProducer
     * methodSize : 1
     */
    public function drainHandler(/*1*/$arg0)
    {

    }
    /**
     * (Memory t)
     * @return Void
     * ()
     * @return Void
     * methodSize : 2
     */
    public function end(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return MessageProducer
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * (Memory message)
     * @return MessageProducer
     * (Memory message, callable replyHandler)
     * @return MessageProducer
     * methodSize : 2
     */
    public function send(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (int maxSize)
     * @return MessageProducer
     * methodSize : 1
     */
    public function setWriteQueueMaxSize(/*1*/$arg0)
    {

    }
    /**
     * (Memory data)
     * @return MessageProducer
     * methodSize : 1
     */
    public function write(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function writeQueueFull(/*0*/)
    {

    }
}

File: vertx-lang-php-jphp/src/main/api/io/vertx/php/core/eventbus/MessageProducer.java
package io.vertx.php.core.eventbus;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.streams.WriteStream;
import io.vertx.php.core.eventbus.Message;
@Name("MessageProducer")
@Namespace("io\\vertx\\php\\core\\eventbus")
public interface MessageProducer<S extends io.vertx.core.eventbus.MessageProducer<T>, T> extends IWrapper<S>{
    
    @Signature
    default void end(Environment __env__, Memory... args) {
        this.getWrappedObject().end()
    }

    @Signature
    default void end(Environment __env__, Memory... args) {
        this.getWrappedObject().end()
    }
    
    @Signature
    default Memory writeQueueFull(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().writeQueueFull()
    }

    @Signature
    default Memory send(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, MessageProducerImpl::new, this.getWrappedObject().send(message)
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }

    @Signature
    default Memory write(Environment __env__, Memory... args) {
        this.getWrappedObject().write(data)
    }

    @Signature
    default Memory setWriteQueueMaxSize(Environment __env__, Memory... args) {
        this.getWrappedObject().setWriteQueueMaxSize(maxSize)
    }

    @Signature
    default Memory drainHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().drainHandler(handler)
    }

    @Signature
    default Memory deliveryOptions(Environment __env__, Memory... args) {
        this.getWrappedObject().deliveryOptions(options)
    }
    
    @Signature
    default Memory address(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().address()
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }

}
